### PR TITLE
more tests

### LINKED ISSUE #516: Inadequate Test Coverage: Less than 80%
Let's add more tests

### PR DESCRIPTION
**Pull request by yegor256**
closes #516

## Summary by CodeRabbit

* **Tests**
  * Large expansion of test coverage for Sugar→Salty conversion, adding many cases for recursive expressions, binding ordering/preservation, formations/pairs, literals, Unicode, deep nesting, negations and other edge cases.
* **Test Data**
  * Added numerous condition-pack YAML fixtures covering and/or/not/in/part-of/xi/nf/termination scenarios and explicit failure cases.
* **Chores**
  * Expanded public API surface to support additional test and validation scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: test/RuleSpec.hs [hunk 1]
```diff
@@ -29,30 +29,74 @@ data ConditionPack = ConditionPack
...
+      isNF (ExFormation []) ctx `shouldBe` True
+    it "returns true for formation with only delta binding" $
+      isNF (ExFormation [BiDelta (BtMany ["00", "01"])]) ctx `shouldBe` True
+    it "returns true for formation with only void binding" $
+      isNF (ExFormation [BiVoid (AtLabel "x")]) ctx `shouldBe` True
+    it "returns true for formation with only lambda binding" $
+      isNF (ExFormation [BiLambda "Func"]) ctx `shouldBe` True
+    it "returns true for formation with delta void and lambda" $
+      isNF (ExFormation [BiDelta (BtOne "FF"), BiVoid (AtLabel "y"), BiLambda "G"]) ctx `shouldBe` True
+  describe "matchProgramWithRule matches programs" $ do
```
Comment (maxonfjvipon) [inline test/RuleSpec.hs]: @yegor256 this test is unnecessary, we have it in yaml packs

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

File: test/SugarSpec.hs [hunk 2]
```diff
@@ -287,3 +287,150 @@ spec = do
...
+          salty = toSalty cst
+      render salty `shouldContain` "number"
+
+  describe "toSalty handles single named application argument" $
+    it "processes single tau binding correctly" $ do
+      prog <- parseProgramThrows "{Q.x(y -> $)}"
+      let cst = programToCST prog
+          salty = toSalty cst
+          rendered = render salty
+      rendered `shouldContain` "y"
```
[the remark this answers is not in the document]
Comment (maxonfjvipon) [reply test/SugarSpec.hs]: @yegor256 check this one

**Comment by maxonfjvipon** [comment]
@rultor merge
